Add card rarity color scheme and card view

geoffmaddox opened the pull request:
- Adds card rarity calculation
- Adds RPG-style rarity color scheme for cards and inverted negatives
- Updates grid view to use color scheme
- Adds card view to display card information (number, text, prime, square, rarity color)

![image](https://github.com/user-attachments/assets/971e60db-14fb-4be9-9cae-c284afbd97a5)
![image](https://github.com/user-attachments/assets/c17eb951-34b0-463d-b1de-0b6bcd8f6d5f)

**GridView**
[Screencast from 2025-01-25 11-20-45.webm](https://github.com/user-attachments/assets/d5aeceea-ba52-4e94-a617-31d6183b60f9)

**Card views / rarity**
[Screencast from 2025-01-25 11-11-15.webm](https://github.com/user-attachments/assets/a739caee-f0b7-4a0b-9775-2460bd23c89d)

Commit:
Add card rarity color scheme and card view

- Adds card rarity calculation
- Adds RPG-style rarity color scheme for cards and inverted negatives
- Updates grid view to use color scheme
- Adds card view to display card information (number, text, prime, square, rarity color)

jasonhenline commented:
I really like this overall, but I'm worried that it makes the "home" view (numbers 0-99) in the grid look really drab with all the gray. Even now that I've pulled a lot I still pay a lot of attention to my "home" grid page because I'm interested in how many rows and columns I'm completed and in how close I am to completing a page.

I've heard there's kind of a standard for rarity colors in games like this, and I think it's valuable to stick with a visual language that is standard, but I wonder if it would make sense to get rid of the "gray" first rarity color and have it start with green instead to make the "home" page look more appealing.

What do you think?

jasonhenline commented on package.json:
```diff
@@ -45,7 +45,8 @@
     "react-native-vector-icons": "^10.2.0",
     "react-native-web": "~0.19.13",
-    "react-native-webview": "13.12.5"
+    "react-native-webview": "13.12.5",
```
What is this being used for?

geoffmaddox replied:
This line is just adding the comma because I added the `react-number-to-words` after, which just translates the number into words.

jasonhenline replied:
Haha, I've got to get my diff setting right.

geoffmaddox commented:
> I really like this overall, but I'm worried that it makes the "home" view (numbers 0-99) in the grid look really drab with all the gray. Even now that I've pulled a lot I still pay a lot of attention to my "home" grid page because I'm interested in how many rows and columns I'm completed and in how close I am to completing a page.
> 
> I've heard there's kind of a standard for rarity colors in games like this, and I think it's valuable to stick with a visual language that is standard, but I wonder if it would make sense to get rid of the "gray" first rarity color and have it start with green instead to make the "home" page look more appealing.
> 
> What do you think?

Yeah, the traditional Diablo color scheme follows the gray -> green -> blue -> ... progression, but I can definitely see how the gray is a little drab on the homescreen. I can shift the rarity colors down and find a new highest rarity color... maybe some kind of a red since a 500+ is so intense?

jasonhenline commented:
In the grid view I also think it looks a lot better to have bright text on a darker background for the positives. I'm not sure if there's a good way to do that and to still have the card view look nice.

jGowgiel commented:
When I've worked on side project-y stuff like this before I've used paletton.com to try and make color palettes - it's got a bit of sidebar-ad-spam nowadays (gotta make money somehow I guess) but the core of the site is still pretty solid. It goes up to "tetrad" styles, and you can then give it a base color (maybe a favorite shade of green you want) and then the "presets" tab at the bottom lets you really easily hue shift from there to different settings.

Generated off of a base green of `00c400`, there's some palettes like:
* A "dark neon" palette: https://paletton.com/#uid=72P1g0kuiiUvNtru4nlwudexs7O
* A "bright pastel" palette: https://paletton.com/#uid=72P1g0kl1Wx1x+IcEXDsUWkWEVB

If you're fishing around for new colors and tones that are still coherent, and wanted to do something like "regular green and negative green," the provided shades and so on from here might be useful!

geoffmaddox commented:
> In the grid view I also think it looks a lot better to have bright text on a darker background for the positives. I'm not sure if there's a good way to do that and to still have the card view look nice.

That's good feedback, I've tried to do that by using two colors for the rarity: a card color and grid color (where the grid color is the card color darkened a bit to keep it thematic, but also not to be too bright to make white not work). I've also removed the yellow color for now because it's not very distinguishable from the orange (which compresses the upper red ranges). I'm not really sure past green / blue / orange / purple if there is a standard TCG / RPG rarity color scheme.

> If we're fishing around for new colors and tones that are still coherent, and wanted to do something like "regular green and negative green," the provided shades and so on from here might be useful!

This is mostly attempting to leverage the "standard" RPG item rarity scale (green = uncommon, blue = rare, etc.), but they could probably be selected to make them more cohesive from a color theory perspective. I'll have to poke around with that, but I've got something that I think is pretty good, but happy to get feedback.

How does this feel?
[Screencast from 2025-01-25 12-20-17.webm](https://github.com/user-attachments/assets/343a0e6b-680f-40c4-9fc0-1e0e2bf0b27d)

jGowgiel commented:
> This is mostly attempting to leverage the "standard" RPG item rarity scale (green = uncommon, blue = rare, etc.), but they could probably be selected to make them more cohesive from a color theory perspective

Yeah, I basically just tweaked around the Paletton settings until I found a tetrad that was green -> blue -> yellow/orange -> red, to match with that rarity scale (while still having some kind of color theory alignment thanks to the tool which I know next to nothing about in reality). Just figured I'd toss that link out since I've found it useful in the past!

Commit:
Update color scheme

- Updates color scheme to use theory-based colors
- Updates grids to white on dark background color theme
- Shifts color scheme to start at green

geoffmaddox commented:
Okay, based on the feedback, I updated the color scheme to be more consistent with color theory (defined in HSL with consistent contrast and regular points on the color wheel with some complimentary colors), and updated the grid page to use a white font on a darker background, where the negative colors leverage a darker version of the rarity (and green is the common rarity instead of gray).

![image](https://github.com/user-attachments/assets/b7cb0969-ebe5-475d-94c1-b7524e6ff59c)
![image](https://github.com/user-attachments/assets/31e55296-3887-48cf-ac29-e5e3d465c251)
![image](https://github.com/user-attachments/assets/6c22402a-1734-44d5-bab7-55e918d07488)
![image](https://github.com/user-attachments/assets/ed37cf40-c76b-4903-98ff-1de6a9eb0b35)
![image](https://github.com/user-attachments/assets/1ba21a7a-e895-4ee8-9a5c-25bffa686b42)
![image](https://github.com/user-attachments/assets/5bb45e49-67f9-47d0-9fd1-e8dd05f73d38)
![image](https://github.com/user-attachments/assets/a4188917-f168-4524-8ad8-080640e078a2)
![image](https://github.com/user-attachments/assets/1012653b-dcc6-4511-a193-cdf8d18c02d7)

[Screencast from 2025-01-25 14-50-38.webm](https://github.com/user-attachments/assets/cc2ae6f6-b60b-4d5e-a0a4-f49fd1a844b9)

Commit:
Update 0 grid color to a darker gray
